students/minchan/westagram/users/views.py

```diff
@@ -1,3 +1,93 @@
-from django.shortcuts import render
+import json
+import re
+from django.http.response   import JsonResponse
+from django.views           import View
+from users.models           import Users
+from django.http            import HttpResponse
 
 # Create your views here.
+class user_register(View) :
+    def valid_email(self,email) :
+        email_regular       = re.compile("[@][a-zA-Z]*[.]")
+        if email_regular.search(email) == None :
+            return False
+        return True
+
+
+    def is_email_duplicate(self,email) :
+        if Users.objects.filter(email=email) :
+            return True # 중복값이 있으면 True를 반환.
+        return False
+    
+
+    def valid_pw(self,pw) :
+        # 검사항목 - 0: 8자리 이상, 1:문자포함, 2:숫자포함, #:특수문자포함
+        test_case           = [0,0,0,0]
+        alphabet_regular    = re.compile("[a-zA-Z]")
+        number_regular      = re.compile("[\d]")
+        symbol_regular      = re.compile("[`~!@#$%^&*(),<.>/?]") #더 많은 특수문자의 허용은 [^0-9a-zA-Z]로 대체 가능.
+        #0 길이검사.
+        if len(pw) >= 8 : 
+            test_case[0] = 1
+
+        #1. 알파벳 포함 검사
+        for i in pw :
+            if alphabet_regular.match(i) != None :
+                test_case[1] = 1
+                break
+
+        #2. 숫자 포함 검사
+        for i in pw :
+            if number_regular.match(i) != None :
+                test_case[2] = 1
+                break
+
+        #3. 특수문자 포함 검사
+        for i in pw :
+            if symbol_regular.match(i) != None :
+                test_case[3] = 1
+                break
+
+        return test_case  
+
+
+    def post(self,request) :
+        data                = json.loads(request.body)
+        data_email          = data["email"]
+        data_pw             = data["password"]
+        wrong_pw_message    = ""
+        # 검사항목 - 0: 8자리 이상, 1:문자포함, 2:숫자포함, #:특수문자포함
+        wrong_pw_list       = ["길이가 8 이하입니다.", "영문이 포함되어있지 않습니다.", "숫자가 포함되어 있지 않습니다.", "특수문자가 포함되어 있지 않습니다."]
+        password_check      = self.valid_pw(data_pw)
+        
+        #1. 이메일, 패스워드 있는지 확인하기.
+        if data_email == None or data_pw == None :
+            return JsonResponse({"MESSAGE": "KEY_ERROR"}, status=400)
```

Comment:
주석 없이도 이해할 수 있는 코드가 좋은 코드라고 생각해주세요~
KeyError를 케치하기 위함이라면 아래와 같이 작성하는게 가독성이 좋을 것 같습니다.

```suggestion
        if not (email or password):
            return JsonResponse({"MESSAGE": "KEY_ERROR"}, status=400)
```

Reply:
주석 모두 지웠습니다. 함수 명으로 충분한 것 같습니다.